src/building/caravanserai.c

```diff
@@ -70,3 +94,51 @@ int building_caravanserai_get_storage_destination(building *caravanserai)
     }
     return 0;
 }
+
+void window_building_draw_caravanserai_action(building_info_context *c)
```

Comment:
I know you changed the code position to consolidate the whole "caravanserai" code, but UI code shouldn't be inside `src/building`. If you want to consolidate it better, create a `src/window/trade_policy.c` and move the UI code there.